src/dbnode/generated/thrift/rpc.thrift

```diff
@@ -312,6 +314,41 @@ struct HealthResult {
 	2: required string status
 }
 
+enum AggregateQueryType {
+	AGGREGATE_BY_TAG_NAME,
+	AGGREGATE_BY_TAG_NAME_VALUE,
+}
+
+// AggregateQueryRequest comes from our desire to aggregate on incoming data.
+// We currently only support retrieval of facets, but could extend this based on
+// requirements in the future. Currently, the two predominant use-cases are:
+// (1) Given a filter query (optionally), return all known tag keys matching this restriction
+// (2) Given a filter query (optionally), return all know tag key+values matching this restriction
+struct AggregateQueryRequest {
+	1: required Query query
+	2: required i64 rangeStart
+	3: required i64 rangeEnd
+	4: required string nameSpace
+	5: optional i64 limit
+	6: optional AggregateQueryType aggregateQueryType = AggregateQueryType.AGGREGATE_BY_TAG_NAME_VALUE
+	7: optional TimeType rangeType = TimeType.UNIX_SECONDS
+	8: optional TimeType resultTimeType = TimeType.UNIX_SECONDS
+}
+
+struct AggregateQueryResult {
+	1: required list<AggregateQueryResultTagNameElement> results
+	2: required bool exhaustive
+}
+
+struct AggregateQueryResultTagNameElement {
+	1: required string tagName
+	2: optional list<AggregateQueryResultTagValueElement> tagValues
+}
+
+struct AggregateQueryResultTagValueElement {
+	1: required string tagValue
```

Comment:
Guessing thrift doesn't allow byte slices instead?

Reply:
Fair, will add performant/non-performant types using the `...Raw` convention.